Proposals/TrustedAppsReplacement.md

```diff
@@ -0,0 +1,397 @@
+    acl:allowedClient   <https://moralapp.com/card#i>,
+                        <https://goodguys.org/card#i>;
+    acl:allowedIDP      <https://trustworthyidp.com>.
+```
+
+*TagAC*
+Let us assume that the goal is now to only allow the good actors to access all files tagged with `https://organization.org/tags/sprintPlanning` and that "meeting1.ttl" is tagged as "sprintPlanning".
+```
+@prefix  acl:  <http://www.w3.org/ns/auth/acl#>.
+
+<#authorization>
+    a                   acl:Authorization;
+    acl:accessTo        <https://mypod.com/>;
+    acl:tag             <https://organization.org/tags/sprintPlanning>;
+    acl:mode            acl:Read;
+    acl:agentClass      acl:AuthenticatedAgent;
+    acl:bannedClient    acl:AuthenticatedAgent; # Ban all agents be default
+    acl:bannedIDP       "*"^^xsd:string;
+    acl:allowedClient   <https://moralapp.com/card#i>,
+                        <https://goodguys.org/card#i>;
+    acl:allowedIDP      <https://trustworthyidp.com>.
+```
+
+
+### General Networking Flow
+
+Below is how token procurement and resquest process works.
+
+Again, it is important to stress that a resource server can ensure that a token is signed by a certain identity provider, but cannot be completely sure that a token is used by a certain application. If the application and identity provider are in collusion, the token can be spoofed.
+
+In the following scenario, Alice is going to go to `https://badguys.com` not knowing that it's a malicious site. Bob, however, does know about badguys, and has set up access control rules to prevent badguys from being able to take his data.
+
+This deviates from the current dpop flow in the following ways:
+
+AUTHORIZATION
+ - In step 7, an additional field is provided that defines the webid of the app
+ - In step 8, an app webid is required
+ - In step 9, the redirects in the app webid are validated
+ - In step 11, the client_id is embedded in the verifiable credential
+
+SENDING REQUEST
+ - In step 9, we extract wthe client id
+ - In step 10, we fail because of the applied access control rules
+
+![](https://i.imgur.com/k1yymTI.png)
+
+
+#### Authorization
+
+##### 1. Alice navigates to badguys.com
+Alice goes to badguys.com, not knowing that this is an app that will malicously use data.
+
+##### 2. Alice Selects her WebId
+Alice tells badguys.com her WebId and it is fetched. Below is Alice's WebId:
+```
+@prefix : <#>.
+@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+@prefix n0: <http://xmlns.com/foaf/0.1/>.
+@prefix schem: <http://schema.org/>.
+
+:me
+    a schem:Person, n0:Person;
+    solid:oidcIssuer <https://idp.com>,
+                      <https://otheridp.com>.
+```
+The important thing to note here is the `solid:oidcIssuer` that links a variable number of issuers to Alice. All these issuers are trusted by Alice, but may not be trusted by other parties. In the case that a party does not trust one issuer, Alice should use a different one.
```

Comment:
I don't quite follow the intent of solid:oidcIssuer here. The `rdfs:comment` for the term in the Solid vocabulary (https://www.w3.org/ns/solid/terms#oidcIssuer) states: "The preferred OpenID Connect issuer URI for a given Web ID", so what's the reasoning behind having multiple here? Is the intent here to allow Alice to list all the IdP's that she personally trusts (which might not even include the IdP that she uses to login)? If so, then maybe it should be a separate predicate like `solid:trustedOidcIssuer`?

Reply:
Perhaps the `rdfs:comment` should be changed to say "A trusted OpenID connect issuer URI for a given webid". The discussion on multiple OPs is [here](https://github.com/solid/authorization-and-access-control-panel/issues/59)

Every solid:oidcIssuer is trusted by Alice. Resource Servers MUST reject any tokens that are signed by issuers that are not listed in Alice's WebID as stated in [step 6 of the request sending flow](https://github.com/solid/authorization-and-access-control-panel/pull/61/files#diff-c5f579abe0451cde312e9e6d319c3c6aR376)